tests/mock_crust/utils.rs

```diff
@@ -322,7 +322,7 @@ pub fn create_connected_nodes_with_cache(network: &Network, size: usize, use_cac
     let n = cmp::min(nodes.len(), network.min_section_size()) - 1;
 
     for node in &mut nodes {
-        expect_next_event!(node, Event::Connected);
+        expect_any_event!(node, Event::Connected if true);
```

Comment:
This function is used by many of the other mock crust tests, so it would be good to not have to downgrade the strictness of this call.  Once your subsequent commit comes in (to avoid dropping tunnelled candidates) can this change be reverted?